### ISSUE TITLE
Feature proposal: Convert Entity OneToMany relations into data.

### ISSUE THREAD

**Issue by fider-apptimia** [mikro-orm/mikro-orm#528]
Feature proposal.

I've not found any built-in way to convert entity and all of it's relations into pojo.

Example data:
```typescript
@Entity()
class Item {
  @ManyToOne()
  data!: Data;

  @OneToMany(() => Details, d => d.item)
  details!: Details;
}

@Entity()
class Data {
  @PrimaryKey()
  id!: string;

  @OneToMany(() => Item, i => i.data)
  items = new Collection<Item>(this);
}

let entity: Data = await entityManager.findOne(Data, ...);
```

**How to do it today:**
```typescript
let pojo = {
  ...ent,
  items: ent.items.getItems()
}

let nestedPojo = {
  ...ent,
  items: ent.items.getItems().map( item => ({
    ...item
    details: item.details.getItems()
  }) )
}
```

**How it can be done:**
```typescript
const nestedPojo: RawEntity<Data> = getRaw(entity);
```

**I find above feature/proposal useful especially for data getters - it can save time when mapping multiple nested data structures.**

- Proposed signature of `getRaw` implementation:
```typescript
function getRaw(entity: object, opts: { mappedBy: true }): RawEntity<Data>;
function getRaw(entity: object, opts: { mappedBy: false}): RawEntityUniDir<Data>;
```
- For `mappedBy: false` (default value) inversed side should not contain relation to parent (to owning side).
- `RawEntity` should convert each property of type `Collection<T>` into `T[]` recursively.
- `RawEntityUniDir` should be the same as `RawEntity` but should exclude relation to parent (in all cases when I wanted to use raw data it was redundant).

**Comment by B4nan** [comment]
There is `wrap(entity).toJSON()` for this purpose.

**Comment by B4nan** [comment]
Closing as implemented as part of #531, which is now merged in v4 (`dev` branch). Subscribe here for updates #527.
